packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -0,0 +1,229 @@
+import { Component, Host, h, Prop, Fragment } from '@stencil/core';
+import classnames from 'classnames';
+
+declare const window: Window &
+  typeof globalThis & {
+    dataLayer: any;
+  };
+
+@Component({
+  tag: 'va-pagination',
+  styleUrl: 'va-pagination.css',
+  shadow: true,
+})
+export class VaPagination {
+  /**
+   * Fires when a page is selected
+   */
+  @Prop() pageSelect: any;
+
+  /**
+   * Fires when a page is selected if enable-analytics is true
+   */
+  @Prop() trackEvent: any;
+
+  /**
+   * Aria-label suffix text for buttons
+   */
+  @Prop() ariaLabelSuffix: string = '';
+
+  /**
+   * Whether or not an analytics event will be fired
+   */
+  @Prop() enableAnalytics: boolean = true;
+
+  /**
+   * The maximum number of pages to show at once
+   */
+  @Prop() maxPageListLength: number = 10;
+
+  /**
+   * The current page number
+   */
+  @Prop() page: number;
+
+  /**
+   * The total number of pages
+   */
+  @Prop() pages: number;
+
+  /**
+   * Display last page number when the page count exceeds
+   * `maxPageListLength`
+   */
+  @Prop() showLastPage: boolean;
+
+  private handleTrackEvent = (...args) => {
+    if (!this.trackEvent) {
+      if (!window.dataLayer) return;
+      window.dataLayer.push(...args);
+    } else {
+      this.trackEvent(...args);
+    }
+  };
```

Comment:
I don't think we need to work with the datalayer directly in here, we can just fire `component-library-analytics` like we're doing in other places. `vets-website` [listens for that event](https://github.com/department-of-veterans-affairs/vets-website/blob/aa496bd9876a9f204b367b5b4b4234e52a8f7846/src/platform/site-wide/component-library-analytics-setup.js#L104-L107) and then pushes to `window.dataLayer` as appropriate.

Reply:
Thanks for that link. It seems we only have 2 web components registered there. Do we need to create a ticket to add our other web components using analytics?

Reply:
Possibly. I've tried to let @bsmartin-ep know when we get a new component but we don't really have a formal process, and it's a bit unclear who is responsible for that bit of code.

Reply:
I can take a look at our other web components and make a ticket for things we need to update. I also spoke with @bsmartin-ep this morning and put together some [guidelines](https://vfs.atlassian.net/wiki/spaces/DST/pages/2079817745/Component+development+process#Analytics%5BinlineExtension%5D) on how to handle updating/adding the analytics events for our components